test/shallowbackup.js

```diff
@@ -0,0 +1,140 @@
+// Copyright © 2017 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it */
+'use strict';
+
+const assert = require('assert');
+const backup = require('../includes/shallowbackup.js');
+const fs = require('fs');
+const nock = require('nock');
+
+describe('#unit Perform backup using shallow back', function() {
+  const dbUrl = 'http://localhost:5984/animaldb';
+
+  it('should perform a shallow backup', function(done) {
+    var couch = nock(dbUrl)
+      // batch 1
+      .get('/_all_docs')
+      .query({limit: 3, include_docs: true})
+      .reply(200, JSON.parse(fs.readFileSync('./test/fixtures/animaldb_all_docs_1.json', 'utf8')))
+      // batch 2
+      .get('/_all_docs')
+      .query({limit: 3, include_docs: true, startkey: '"badger\\u0000"'})
+      .reply(200, JSON.parse(fs.readFileSync('./test/fixtures/animaldb_all_docs_2.json', 'utf8')))
+      // batch 3
+      .get('/_all_docs')
+      .query({limit: 3, include_docs: true, startkey: '"kookaburra\\u0000"'})
+      .reply(200, JSON.parse(fs.readFileSync('./test/fixtures/animaldb_all_docs_3.json', 'utf8')))
+      // batch 4
+      .get('/_all_docs')
+      .query({limit: 3, include_docs: true, startkey: '"snipe\\u0000"'})
+      .reply(200, JSON.parse(fs.readFileSync('./test/fixtures/animaldb_all_docs_4.json', 'utf8')));
+
+    backup(dbUrl, 3, 1, null, false)
```

Comment:
maybe call this with `dbUrl, 3, null, null, null` since the args are ignored anyway, this makes it look like they have purpose
and for the later instances of the same call